sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/TextAnalyticsAsyncClient.java

```diff
@@ -1043,21 +1046,51 @@ public String getDefaultLanguage() {
     }
 
     /**
-     * Cancel a long-running operation healthcare task by given job ID.
+     * Cancel a long-running operation healthcare task by given a healthcare task identification number.
      *
      * <p><strong>Code Sample</strong></p>
-     * {@codesnippet com.azure.ai.textanalytics.TextAnalyticsAsyncClient.beginCancelAnalyzeHealthcare#UUID}
+     * {@codesnippet com.azure.ai.textanalytics.TextAnalyticsAsyncClient.beginCancelAnalyzeHealthcare#String-RecognizeHealthcareEntityOptions}
      *
-     * @param jobId A job identification number.
+     * @param healthcareTaskId The healthcare task identification number.
+     * @param options The additional configurable {@link RecognizeHealthcareEntityOptions options} that may be passed
+     * when cancelling healthcare task.
      *
      * @return A {@link PollerFlux} that polls the analyze healthcare operation until it has completed, has failed,
      * or has been cancelled.
      *
      * @throws TextAnalyticsException If analyze operation fails.
+     * @throws NullPointerException If {@code healthcareTaskId} is null.
+     */
+    @ServiceMethod(returns = ReturnType.COLLECTION)
+    public PollerFlux<TextAnalyticsOperationResult, Void> beginCancelAnalyzeHealthcare(String healthcareTaskId,
+        RecognizeHealthcareEntityOptions options) {
+        return analyzeHealthcareAsyncClient.beginCancelAnalyzeHealthcare(healthcareTaskId, options, Context.NONE);
+    }
+
+    // Analyze LRO
+    /**
+     * Analyze tasks, such as, entity recognition, PII entity recognition and key phrases extraction in a list of
+     * {@link TextDocumentInput document} with provided request options.
+     *
+     * See <a href="https://aka.ms/talangs">this</a> supported languages in Text Analytics API.
+     *
+     * <p><strong>Code Sample</strong></p>
+     * {@codesnippet com.azure.ai.textanalytics.TextAnalyticsAsyncClient.beginAnalyze#Iterable-AnalyzeTasksOptions}
+     *
+     * @param documents A list of {@link TextDocumentInput documents} to be analyzed.
+     * @param options The additional configurable {@link AnalyzeTasksOptions options} that may be passed when
+     * analyzing a collection of tasks.
+     *
+     * @return A {@link PollerFlux} that polls the analyze a collection of tasks operation until it has completed,
+     * has failed, or has been cancelled. The completed operation returns a {@link PagedFlux} of
+     * {@link AnalyzeTasksResult}.
+     *
+     * @throws TextAnalyticsException If analyze operation fails.
      * @throws NullPointerException If {@code jobId} is null.
      */
     @ServiceMethod(returns = ReturnType.COLLECTION)
-    public PollerFlux<TextAnalyticsOperationResult, Void> beginCancelAnalyzeHealthcare(UUID jobId) {
-        return analyzeHealthcareAsyncClient.beginCancelAnalyzeHealthcare(jobId, Context.NONE);
+    public PollerFlux<TextAnalyticsOperationResult, PagedFlux<AnalyzeTasksResult>> beginAnalyze(
```

Comment:
This naming is too generic, should we consider renaming?

Reply:
Yes. I can change it to beginAnalyzeTasks().

Reply:
`beginAnalyzeTasks` is still very vague. Is this meant to change and not something that got discussed for this preview?
cc: @maririos @iscai-msft

Reply:
For this release we are going with the autogenerated code so there is no focus on naming.
I do agree that for future releases, this name needs rework